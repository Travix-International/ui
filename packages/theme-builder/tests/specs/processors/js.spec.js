const jsProcessor = require('../../../src/processors/js.js');

describe('JS process', () => {
  it('should process object', () => {
    const doc = {
      another: {
        something: 'bbb'
      },
      do: {
        some: {
          string: 'aaa',
          num: '12px',
          color: '#fff',
          // eslint-disable-next-line no-template-curly-in-string
          template: '${color}${some.string}${another.something}'
        }
      }
    };

    const result = jsProcessor.compile(doc);
    expect(result).toMatchSnapshot();
  });
});
