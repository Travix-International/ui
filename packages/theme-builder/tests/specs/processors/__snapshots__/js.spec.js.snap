exports[`JS process should process object 1`] = `
Object {
  "do": Object {
    "some": Object {
      "color": "#fff",
      "num": "12px",
      "string": "aaa",
    },
  },
}
`;
